[TN-333] Protect Weblogin from automated attacks

fajarism opened the pull request:
[[TN-333] Issue](https://github.com/human-internet/humanid-weblogin/issues/65#issuecomment-2813623679)

This PR fixes the TN-333 ticket by adding several features below in the core service:
1. Rate limiter by using [express-rate-limit](https://github.com/express-rate-limit/express-rate-limit). This will limit incoming request to the `request otp` endpoint by matching the ip query params.
2. AES decryption, to decrypt an encrypted `ip` query params used by rate limiter above.
3. AES_SECRET_KEY env value

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced rate limiting for OTP request submissions to enhance security and prevent abuse.
  - Added support for decrypting AES-256-GCM encrypted data within the application.

- **Chores**
  - Added a new environment variable example for AES secret key configuration.
  - Updated dependencies to include rate limiting support.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
feat: add rate limiter for otp request, aes encryption, and env params

Commit:
fix: change encryption algorithm

Commit:
fix: remove iv

falhar commented on src/components/common.js:
```diff
@@ -252,6 +254,38 @@ const parsePhone = (phoneStr, options = {}) => {
+    const iv = raw.subarray(0, ivLength);
+    const tag = raw.subarray(ivLength, ivLength + tagLength);
+    const ciphertext = raw.subarray(ivLength + tagLength);
+    const key = Buffer.from(process.env.AES_SECRET_KEY, "hex");
```
use `config.AES_SECRET_KEY` since you already define up above

fajarism replied:
fixed in the latest commit

Lunasocute commented on src/server/rate-limit.js:
```diff
@@ -0,0 +1,57 @@
+                code: "GENERAL_ERROR",
+                message: "Something went wrong",
+            });
+        };
```
why use response 200 here?

fajarism replied:
Fixed in the latest commit

Commit:
fix: change status code to 400 and change aes_key read to read from config

fajarism commented:
> [Coderabbit](https://github.com/human-internet/humanid-core/pull/91#discussion_r2083526411): Use appropriate HTTP status code for rate limiting

Changed to 400 as the ticket requirement.

> [Firman](https://github.com/human-internet/humanid-core/pull/91#discussion_r2083531188): use config.AES_SECRET_KEY since you already define up above

Changed reading `AES_SECRET_KEY` to `config.AES_SECRET_KEY`, instead of `process.env.AES_SECRET_KEY`

Commit:
fix: throw error when failed to decrypt ip
